api/query/cache/index/filter.go

```diff
@@ -144,6 +151,26 @@ func (c Count) Filter(t TestID) bool {
 	return matches == c.count
 }
 
+// Filter interprets a Link as a filter function over TestIDs.
+func (l Link) Filter(t TestID) bool {
+	name, _, err := l.tests.GetName(t)
+	if err != nil {
+		return false
+	}
+
+	urls, isExist := l.metadata[name]
```

Comment:
nit: Remane `isExist` to `ok` (standard name for the second param in a map lookup)